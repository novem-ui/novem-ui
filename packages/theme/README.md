# `@novem-ui/theme`

> TODO: description

## Usage

```
const theme = require('@novem-ui/theme');

// TODO: DEMONSTRATE API
```
