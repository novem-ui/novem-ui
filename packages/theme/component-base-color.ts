export { ComponentBaseColorProps, ComponentBaseColorProvider, useComponentBaseColor } from './src/component-base-color'
