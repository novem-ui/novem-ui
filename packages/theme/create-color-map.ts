export { default } from './src/utils/create-color-map'
