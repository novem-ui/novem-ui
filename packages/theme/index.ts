export { default as useTheme } from './use-theme'
export { default as withTheme } from './with-theme'
export { default as createColorMap } from './create-color-map'
export { default as ThemeProvider } from './theme-provider'
export { default, Theme, BaseColor, Color, HeadingSizeValue, SpacingValue, TextSizeValue, Shadows } from './theme'
