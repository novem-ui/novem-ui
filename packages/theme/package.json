{
  "name": "@novem-ui/theme",
  "version": "0.0.2-alpha.0",
  "description": "Theme for novem-ui ",
  "author": "Leonel Vieyra <me@leonelv.com>",
  "homepage": "https://github.com/jdrets/novem-ui/tree/main/packages/theme#readme",
  "license": "ISC",
  "main": "index.js",
  "publishConfig": {
    "access": "public",
    "directory": "../../dist/theme"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/jdrets/novem-ui.git"
  },
  "scripts": {
    "test": "jest --env=jsdom --passWithNoTests",
    "build": "yarn build:esm && yarn build:types && yarn build:package-json",
    "build:esm": "../../node_modules/.bin/babel . --root-mode upward --extensions .ts,.tsx -d ../../dist/theme --source-maps",
    "build:types": "tsc -p ./tsconfig.types.json",
    "build:package-json": "../../scripts/create-package.js"
  },
  "bugs": {
    "url": "https://github.com/jdrets/novem-ui/issues"
  },
  "peerDependencies": {
    "@emotion/react": "^11.1.5",
    "@emotion/styled": "^11.1.5",
    "polished": "^4.1.3",
    "react": "^17.0.2",
    "react-dom": "^17.0.2"
  }
}
