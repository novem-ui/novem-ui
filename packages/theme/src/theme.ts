import colors from './colors'
import * as typography from './typography'
import spacing from './spacing'
import { WeightedColorKey } from '../types'

const theme = {
  colors,
  typography,
  spacing,
}

export type BaseColor = `${keyof typeof colors.palette}`
export type Color = `${keyof typeof colors.palette}.${WeightedColorKey}`
export type HeadingSizeValue = keyof typeof typography.headingFontSizes
export type SpacingValue = keyof typeof spacing
export type TextSizeValue = keyof typeof typography.textFontSizes
export type Theme = typeof theme


export default theme
