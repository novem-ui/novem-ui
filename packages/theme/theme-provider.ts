export { default } from './src/theme-provider'
