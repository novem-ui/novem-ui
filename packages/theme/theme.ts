export { default, Theme, BaseColor, Color, HeadingSizeValue, SpacingValue, TextSizeValue } from './src/theme'
