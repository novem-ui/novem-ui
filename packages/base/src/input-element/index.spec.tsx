import React from 'react'
import { render } from '@testing-library/react'
import styled from '@emotion/styled'
import theme from '@novem-ui/theme'

import inputElement from '.'

const ComponentWithBorderProps = styled.div`
  ${inputElement}
`

describe('spacing', () => {
  test('should create a css fragment the correct styles', () => {
    const { container } = render(<ComponentWithBorderProps theme={theme} />)

    expect(container).toMatchSnapshot()
  })

  test('should create a css fragment the correct styles (with custom color)', () => {
    const { container } = render(<ComponentWithBorderProps theme={theme} baseColor="red" />)

    expect(container).toMatchSnapshot()
  })
})
