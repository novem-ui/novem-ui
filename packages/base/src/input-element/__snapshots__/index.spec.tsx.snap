// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`spacing should create a css fragment the correct styles (with custom color) 1`] = `
.emotion-0 {
  border: 1px solid rgba(0, 0, 0, .25);
  border-radius: 1rem;
  box-sizing: border-box;
  height: 3rem;
  font-size: 16px;
  padding: 1rem;
  -webkit-transition: border ease-in 0.25s;
  transition: border ease-in 0.25s;
}

.emotion-0:focus,
.emotion-0:hover {
  outline: unset;
  border-color: #EB5757;
  box-shadow: 0 2px 4px 0 rgba(235,87,87,0.15);
}

.emotion-0:disabled {
  pointer-events: none;
  background-color: rgba(154,152,151,0.15);
}

.emotion-0:-moz-read-only {
  background-color: white;
  border-color: white;
  margin-left: -2px;
  padding: 0.5rem;
  pointer-events: none;
}

.emotion-0:read-only {
  background-color: white;
  border-color: white;
  margin-left: -2px;
  padding: 0.5rem;
  pointer-events: none;
}

<div>
  <div
    class="emotion-0 emotion-1"
  />
</div>
`;

exports[`spacing should create a css fragment the correct styles 1`] = `
.emotion-0 {
  border: 1px solid rgba(0, 0, 0, .25);
  border-radius: 1rem;
  box-sizing: border-box;
  height: 3rem;
  font-size: 16px;
  padding: 1rem;
  -webkit-transition: border ease-in 0.25s;
  transition: border ease-in 0.25s;
}

.emotion-0:focus,
.emotion-0:hover {
  outline: unset;
  border-color: #F1694B;
  box-shadow: 0 2px 4px 0 rgba(241,105,75,0.15);
}

.emotion-0:disabled {
  pointer-events: none;
  background-color: rgba(154,152,151,0.15);
}

.emotion-0:-moz-read-only {
  background-color: white;
  border-color: white;
  margin-left: -2px;
  padding: 0.5rem;
  pointer-events: none;
}

.emotion-0:read-only {
  background-color: white;
  border-color: white;
  margin-left: -2px;
  padding: 0.5rem;
  pointer-events: none;
}

<div>
  <div
    class="emotion-0 emotion-1"
  />
</div>
`;
